E0379 needs to be updated to new format

From: src/test/compile-fail/const-fn-mismatch.rs

Error E0379 needs to a span_label, updating it from:

```
error[E0379]: trait fns cannot be declared const
  --> src/test/compile-fail/const-fn-mismatch.rs:23:5
   |
23 |     const fn f() -> u32 { 22 } //~ ERROR E0379
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^
```

to:

```
error[E0379]: trait fns cannot be declared const
  --> src/test/compile-fail/const-fn-mismatch.rs:23:5
   |
23 |     const fn f() -> u32 { 22 } //~ ERROR E0379
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ trait tns cannot be const
```

Bonus: shrink the span to focus on the `const` part:

```
error[E0379]: trait fns cannot be declared const
  --> src/test/compile-fail/const-fn-mismatch.rs:23:5
   |
23 |     const fn f() -> u32 { 22 } //~ ERROR E0379
   |     ^^^^^ trait tns cannot be const
```

--- a/src/librustc_typeck/check/mod.rs
+++ b/src/librustc_typeck/check/mod.rs
@@ -847,7 +847,9 @@ fn check_trait_fn_not_const<'a,'tcx>(ccx: &CrateCtxt<'a, 'tcx>,
             // good
         }
         hir::Constness::Const => {
-            span_err!(ccx.tcx.sess, span, E0379, "trait fns cannot be declared const");
+            struct_span_err!(ccx.tcx.sess, span, E0379, "trait fns cannot be declared const")
+                .span_label(span, &format!("trait fns cannot be const"))
+                .emit()
         }
     }
 }

--- a/src/test/compile-fail/const-fn-mismatch.rs
+++ b/src/test/compile-fail/const-fn-mismatch.rs
@@ -20,7 +20,9 @@ trait Foo {
 }
 
 impl Foo for u32 {
-    const fn f() -> u32 { 22 } //~ ERROR E0379
+    const fn f() -> u32 { 22 }
+    //~^ ERROR E0379
+    //~| NOTE trait fns cannot be const
 }
 
 fn main() { }

